utils/make-glossary.R

```diff
@@ -1,16 +1,20 @@
 #!/usr/bin/env Rscript
 
 # Create .rda version of glossary for R package.
```

Comment:
for this, personally, I would only fetch the yml file and put it in a `inst/glosario` folder instead of making it a binary data file. Then, when reading the glossary locally, you can directly call `yaml::read_yaml` on `system.file("glosario/glossary.yml", package = "glosario")`. This would have the advantage of providing people with a plain text version of the file if they wanted to inspect it (also easier to diff if needed), and it would make the code in the package simpler because we could use the same code to parse from the web or from the package directly. But maybe there is a reason you chose this approach?

Reply:
I would also agree with this, especially if otherwise a binary is being updated every day on the github repo, which will make it grow and grow. 

One of the things that I would advocate for is using an application directory instead for offline use, which comes in both R and python flavors (see [{rappdirs}](https://cran.r-project.org/web/packages/rappdirs/index.html) and [appdirs](https://github.com/ActiveState/appdirs)). This way, people would be able to periodically update their dictionary without needing to update the package itself.

Reply:
I think we could have both? 
Update the content of the `inst/` folder daily with GitHub Actions so there is always a fresh copy of the glossary with the package, and create a function "update_local_glossary()` (or something like it) that would update the glossary in the application directory.
This feature seems however non-critical for the first release and could be implemented later.

Reply:
> This feature seems however non-critical for the first release and could be implemented later.

Agreed. My thoughts are largely centered around the patterns I see from R users and reproducibility: 

1. people don't update their packages that often or don't know how to update their packages.
2. there is no clear indicator of what version of the glossary people have on their machines, so if it defaults to the one in the package, then definitions that exist in the global glossary may be missing in their local version.
3. if {glosario} is released on CRAN, it will be updated every two months (as per CRAN's policy) at most, but the main glosario repository will be constantly updating. 

These situations mean that if Belle installs the package on March 4th and Sebastian installs the package on July 17th, they will have two different versions of the glossary on their machines. Let's say they contribute a few new definitions to the main glossary on July 16th, but neither of them see these definitions on their lesson because the package was only pushed to CRAN on July 1st. 

I think it would be good to consider these situations before we release this to CRAN and coordinate with the python implementation so that we can reduce the friction that users see.

Reply:
These are good points @zkamvar! and made me think that we might want to consider including some type versioning inside the YAML (a hash of the glossary content? a Git SHA1? an increasing counter? a datetime stamp?) so people/machines could compare versions?

Reply:
I'm going to open a new issue for these discussions